### ISSUE TITLE
Review request: surrounded by

### ISSUE THREAD

**Issue by cmungall** [oborel/obo-relations#308]
The GO intends to add many logical axioms to cell components for spatial relations, based on information curated in Reactome: https://github.com/geneontology/go-ontology/issues/16785

For this we intend to use the surrounded by relation, which is tagged with pending final vetting
http://purl.obolibrary.org/obo/RO_0002219

This relation is heavily in use in Uberon but not yet in use in GO

I think we need to add some verbiage to deal with granularity, organelles are not impenetrable balls, they have openings if you go to a sufficient granularity.

cc @phismith it may be useful to have review from people in the geospatial reasoning community

**Comment by JansenLudger** [comment]
OntoBee: "X surrounded_by y if and only if 
(1) x is adjacent to y and for every region r that is adjacent to x, r overlaps y 
(2) the shared boundary between x and y occupies the majority of the outermost boundary of x"

Is there an implicit AND between (1) and (2)? Or an OR?
(1) seems to imply (2), and thus seems to be redundant if there is an AND between (1) and (2). If there is an OR, (1) seems to be redundant. 
In OntoBee, I did not find any info about the domain and range restrictions. One might want to say that a 2d region X is surrounded by another 2d region Y, although Y does not surround X over and below the plane in which X and Y are situated.

**Comment by ukemi** [comment]
Since we consider things like nuclear pores to be part_of the nuclear envelope, doesn't this definition still describe the nuclear envelope? If we go to sufficient granularity nothing is impenetrable, there is space between molecules. I think we have to consider some type of fiat boundaries for surrounding structures and take those boundaries into consideration when defining the surrounding structure, such as including the pore as part of the envelope.

**Comment by diatomsRcool** [comment]
Has there been a discussion about whether or not this refers to being surrounded as in how a fence surrounds a yard or as in how chocolate surrounds the cream of a Cadbury egg?

**Comment by wdduncan** [comment]
Do y form the boundary of a (BFO) site that is occupied by x?

**Comment by cmungall** [comment]
@JansenLudger - it is meant to say AND. Not quite redundant as (1) encompasses the inverse "surrounds". But open to more elegant formulations.

I'm embarrassed to say I have lost some of the provenance for these relations, they arose in part out of discussions with Brandon Bennet about 2D relations for ENVO and GAZ, am looking through archives...

**Comment by cmungall** [comment]
@diatomsRcool yes, cadbury's egg = cell, chocolate = plasma membrane, cream = cytoplasm. Maybe the wrapping is a kind of cell wall. We can extend this analogy to Kinder surprise eggs and nuclei too. Although I am not sure if Kinder surprise eggs arose from an ancient endocytosis event.

**Comment by cmungall** [comment]
@wdduncan yes I believe so. Although we should be clear if the domain is IC or ME, I think it should be IC in which case your statement only holds if occupied-by is reflexive (ie a site can be surrounded by a ME, and the site occupies itself)

**Comment by cmungall** [comment]
@ukemi agreed

**Comment by ukemi** [comment]
I agree with the egg example, but have a few concerns about the fence. Unless we are viewing the yard as only 2-dimensional and the fence as essentially a line around the 2-dimensional yard. Last week we looked at surrounded_by, bounding_layer_of and lumen_of with respect to organelles. From our interpretation the bounding layer must be a laminar structure and is part_of the object it is 'binding'. So for something like a vacuole, the vacuole is surrounded by cytosol, the vacuolar lumen is lumen_of vacuole. The bounding layer would be the vacuolar membrane. Would it be correct to say the membrane is surrounded by the cytosol? I think it fits the above definition.

**Comment by wdduncan** [comment]
@cmungall In BFO, sites are immaterial entities. What does it meant to say that a site occupies itself? It seems similar to questions as to whether the part of relation should be reflexive. I think there is pragmatic reason for making such relations reflexive: it allows you to more easily query an entity all its parts.

**Comment by cmungall** [comment]
@ukemi yep, we already have bounding-layer-of https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FRO_0002007 which includes laminarity in its definition is used in GO. So the general pattern is

 * X membrane bounding-layer-of X (implies part-of X)
 * X lumen surrounded-by X membrane
 * X surrounded-by {cytosol, whatever the stuff outside is, ...}

Discussion of 2D counterparts: #297 - let's keep fences out of this discussion as potentially confusing. Simulated egg confectionary and cells OK though.

**Comment by nlharris** [comment]
What's the status of this?

**Comment by dosumis** [comment]
I think it would be sufficient to add a comment noting that surrounded by does not imply hermetically sealed X surrounds Y even where X has small openings in it.  The comment could reference nuclear pores as an example.

**Comment by nlharris** [comment]
Add a comment where? Could you do that, as a PR?
(BTW, isn't it like 2:30am where you are??)

**Comment by dosumis** [comment]
rdfs:comment on relation.  (it is 2:30am so I should probably go back to bed rather than make a PR right now).

**Comment by cmungall** [comment]
Comments sounds good!

**Comment by JansenLudger** [comment]
Pores seem to contradict clause (1) of the definition.
